src/App.tsx

```diff
@@ -15,48 +17,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+const getReorderedGoods = function (
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a:string, b: string): number => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reverse = () => {
+    setReverse(!isReversed);
+  };
+
+  const handleSortByAlpabet = () => {
+    setSortType(SortType.ALPABET);
+  };
+
+  const handleSortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const NONE = () => {
+    setSortType(SortType.NONE);
+  };
+
+  const preparedGoods
+  = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+  const condition = (sortType !== SortType.NONE || isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classnames(
+            'button is-info', {
+              'is-light': sortType !== SortType.ALPABET,
+            },
+          )}
+          onClick={handleSortByAlpabet}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classnames(
+            'button is-success', {
+              'is-light': sortType !== SortType.LENGTH,
+            },
+          )}
+          onClick={handleSortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classnames(
+            'button is-warning', {
+              'is-light': sortType !== SortType.NONE,
+            },
+          )}
+          onClick={reverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {condition
```

Comment:
it is not obvious from name what is the condition
use `shouldRenderResetButton` or smth like that